File: Practical-01/C code/bubble_sort.c
#include <stdio.h>

void swap(int *a, int *b) {
    int temp = *a;
    *a = *b;
    *b = temp;
}

int main(){
    int arr[100];
    int n;
    printf("Enter the number of elements: ");
    scanf("%d", &n);

    printf("Enter the array elements:\n");
    for(int i = 0; i < n; i++){
        scanf("%d", &arr[i]);
    }

    int i , j ;
    for(i = 0; i < n-1; i++){
        for(j = 0; j < n-i-1; j++){
            if(arr[j] > arr[j+1]){
                swap(&arr[j], &arr[j+1]);
            }
        }
    }
    printf("Sorted array: \n");
    for(i = 0; i < n; i++){
        printf("%d ", arr[i]);
    }
    return 0;
}

File: Practical-01/Python code/bubble_sort.py
arr = []
print("Enter the number of elements in the array:")
n = int(input())
print("Enter the elements of the array:")
for _ in range(n):
    arr.append(int(input()))

for i in range(n):
    for j in range(0, n-i-1):
        if arr[j] > arr[j+1]:
            arr[j], arr[j+1] = arr[j+1], arr[j]
            print(f"Swapped {arr[j]} and {arr[j+1]}: {arr}")
print("Sorted array is:")
for i in range(n):
    print(arr[i], end=" ")